query_base.go

```diff
@@ -1096,3 +1096,252 @@ func (q cascadeQuery) appendCascade(fmter schema.Formatter, b []byte) []byte {
 	}
 	return b
 }
+
+//------------------------------------------------------------------------------
+
+type idxHintsQuery struct {
+	use    *indexHints
+	ignore *indexHints
+	force  *indexHints
+}
+
+type indexHints struct {
+	names      []schema.QueryWithArgs
+	forJoin    []schema.QueryWithArgs
+	forOrderBy []schema.QueryWithArgs
+	forGroupBy []schema.QueryWithArgs
+}
+
+func (ih *idxHintsQuery) addIdxHints(indexes ...string) []schema.QueryWithArgs {
+	hints := make([]schema.QueryWithArgs, 0, len(indexes))
+	for _, idx := range indexes {
+		hints = append(hints, schema.UnsafeIdent(idx))
+	}
+	return hints
+}
+
+func (ih *idxHintsQuery) appendUseIndex(indexes ...string) {
+	if len(indexes) == 0 {
+		return
+	}
+	if ih.use == nil {
+		ih.use = new(indexHints)
+	}
+	ih.use.names = ih.addIdxHints(indexes...)
+}
```

Comment:
The func name `appendUseIndex` says append, but instead it overwrites `names`. I think this can be simplified like this:

```go
func (ih *idxHintsQuery) appendUseIndex(indexes ...string) {
    use := ih.lazyUse()
    use.names = appendIndexHints(use.names, indexes...)
}

func (ih *idxHintsQuery) lazyUse() *indexHints {
	if ih.use == nil {
		ih.use = new(indexHints)
	}
	return ih.use
}

func appendIndexHints(hints []schema.QueryWithArgs, indexes ...string) []schema.QueryWithArgs {
	for _, idx := range indexes {
		hints = append(hints, schema.UnsafeIdent(idx))
	}
	return hints
}
```

What do you think?